### PR TITLE
Remove kube-rbac-proxy

### PR DESCRIPTION
**Pull request by jayesh-srivastava**
**What type of PR is this?**

**What this PR does / why we need it**:
This PR makes the required changes needed for v1alpha4. According to the v1alpha4 migration documentation, kube-rbac-proxy should be removed (https://cluster-api.sigs.k8s.io/developer/providers/v1alpha3-to-v1alpha4).

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

_Please confirm that if this PR changes any image versions, then that's the sole change this PR makes._

**TODOs**:

- [ ] squashed commits
- [ ] includes documentation
- [ ] adds unit tests

**Release note**:
```
Remove kube-rbac-proxy
```

### CODE REVIEW

**Comment by cpanato** [comment]
@jayesh-srivastava can you explain why we need to remove this in the PR summary? also when it is ready can you remove the WIP? thanks!

File: config/manager/manager.yaml [hunk 1]
```diff
@@ -21,7 +21,7 @@ spec:
       containers:
       - args:
         - --leader-elect
-        - "--metrics-bind-addr=127.0.0.1:8080"
+        - "--metrics-bind-addr=localhost:8080"
```
Comment (cpanato) [inline config/manager/manager.yaml:24]: i think this should stay as `127.0.0.1`

**Review by cpanato — APPROVED** [review]
/lgtm
/approve

**Comment by jayesh-srivastava** [comment]
@cpanato guess there shouldn't be `do-not-merge/hold` label

**Pull request merged**